server/sonar-server/src/main/java/org/sonar/server/organization/ws/RemoveMemberAction.java

```diff
@@ -98,6 +98,7 @@ private void removeMember(DbSession dbSession, OrganizationDto organization, Use
     String organizationUuid = organization.getUuid();
     dbClient.userPermissionDao().deleteOrganizationMemberPermissions(dbSession, organizationUuid, userId);
     dbClient.permissionTemplateDao().deleteUserPermissionsByOrganization(dbSession, organizationUuid, userId);
+    dbClient.qProfileEditUsersDao().deleteByUser(dbSession, user);
```

Comment:
shouldn't it be restricted to an organization? When you remove a member, you want to delete qprofile user permissions only for the given organization. Either use a `deleteByOrganizationAndUser` or get the organization profile to delete